### PR TITLE
Bump to GNU m4 @ 1.4.19

### PR DESCRIPTION
**Pull request by jesec**
```
This stable release is a minor update, collecting several years of
portability improvements, as well as a couple of minor optimizations
to performance.
```

Notably, GNU m4 1.4.19 has a newer gnulib that has much better compatibility. 

Fixes compilation errors we encountered on some platforms. 

```
external/org_gnu_m4/lib/strsignal.c:108:25: error: use of undeclared identifier 'sys_siglist'
      || (desc = INTUSE(_sys_siglist)[signum]) == NULL)
                        ^
external/org_gnu_m4/lib/strsignal.c:67:24: note: expanded from macro '_sys_siglist'
#  define _sys_siglist sys_siglist
```

```
bazel-out/host/bin/external/org_gnu_m4/_objs/m4/clean-temp.o:clean-temp.c:function create_temp_dir: error: undefined reference to '__muloti4'
bazel-out/host/bin/external/org_gnu_m4/_objs/m4/fatal-signal.o:fatal-signal.c:function at_fatal_signal: error: undefined reference to '__muloti4'
bazel-out/host/bin/external/org_gnu_m4/_objs/m4/symtab.o:symtab.c:function symtab_init: error: undefined reference to '__muloti4'
```

Related: #40, https://github.com/hdl/bazel_rules_hdl/pull/24#discussion_r544643490

### CODE REVIEW

**Comment by per-gron** [comment]
I'm sorry if you don't have access to the build logs. The error is this:

```
ERROR: /builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/external/org_gnu_m4/BUILD.bazel:53:10: Linking of rule '@org_gnu_m4//:m4' failed (Exit 1) clang failed: error executing command external/llvm_toolchain/bin/clang @bazel-out/host/bin/external/org_gnu_m4/m4-2.params

Use --sandbox_debug to see verbose messages from the sandbox
ld.lld: error: undefined symbol: posix_spawn_file_actions_addchdir_np
>>> referenced by spawn_faction_addchdir.c
>>>               bazel-out/host/bin/external/org_gnu_m4/_objs/m4/spawn_faction_addchdir.o:(posix_spawn_file_actions_addchdir)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by jesec** [comment]
> I'm sorry if you don't have access to the build logs. The error is this:
> 
> ```
> ERROR: /builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/external/org_gnu_m4/BUILD.bazel:53:10: Linking of rule '@org_gnu_m4//:m4' failed (Exit 1) clang failed: error executing command external/llvm_toolchain/bin/clang @bazel-out/host/bin/external/org_gnu_m4/m4-2.params
> 
> Use --sandbox_debug to see verbose messages from the sandbox
> ld.lld: error: undefined symbol: posix_spawn_file_actions_addchdir_np
> >>> referenced by spawn_faction_addchdir.c
> >>>               bazel-out/host/bin/external/org_gnu_m4/_objs/m4/spawn_faction_addchdir.o:(posix_spawn_file_actions_addchdir)
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> ```

Thanks. I do have to access to the logs.

Interestingly, the platform CI uses seems to be weirder than ours. 😂

File: dependency_support/org_gnu_m4/bundled.BUILD.bazel [hunk 1]
```diff
@@ -40,66 +41,134 @@ cc_library(
...
-    includes = ["lib", "build-aux/snippet"],
+    includes = ["lib"],
     visibility = ["//visibility:private"],
 )
 
 cc_binary(
     name = "m4",
-    srcs = glob(
-        ["lib/*.c", "src/*.c"],
-        exclude = HEADERS_WITH_C_EXTENSION + [
```
Comment (per-gron) [inline dependency_support/org_gnu_m4/bundled.BUILD.bazel:57]: I think these exclusions are a good thing to have. IIRC a normal m4 build (using its own build system) does not include all of those files. The m4 source distro has a lot of reimplementations of stuff that is present in any sane Linux. This is probably why you're seeing build errors.

I don't really have an opinion about how to do this though, as long as the build works.

**Review by per-gron — CHANGES_REQUESTED** [review]

**Comment by per-gron** [comment]
> Interestingly, the platform CI uses seems to be weirder than ours. 😂

😄 

If it helps you: This is where the Cloud Build OS is specified https://github.com/hdl/bazel_rules_hdl/blob/main/cloudbuild.yaml#L36

File: dependency_support/org_gnu_m4/bundled.BUILD.bazel (hunk 1 above)
Comment (jesec) [reply dependency_support/org_gnu_m4/bundled.BUILD.bazel:57]: Oh. I switched from the "blocklist" (exclusion) to "allowlist" approach, so actually less files are included. 

that's indeed why I am seeing build errors, though. 

(if you compare the first revision abd0aac2fbb90adc269d709d4bf2c86d127b95ae with the second one eaf226bd4c3ecec8714cbb763b5f2f7d8ffe48b8, you will see that `lib/reallocarray.c` has been added)

**Comment by jesec** [comment]
I decided to switch to a different approach that benefits all GNU tools, and will open a new PR soon. Closing this PR.

**Pull request closed without merging**